uPortal-api/uPortal-api-internal/src/test/java/org/apereo/portal/portlets/account/UserAccountHelperTest.java

```diff
@@ -0,0 +1,124 @@
+/**
+ * Licensed to Apereo under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright ownership. Apereo
+ * licenses this file to you under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the License at the
+ * following location:
+ *
+ * <p>
+ *
+ * <p>http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apereo.portal.portlets.account;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.*;
+import org.apereo.portal.persondir.ILocalAccountDao;
+import org.apereo.portal.persondir.ILocalAccountPerson;
+import org.apereo.portal.portletpublishing.xml.Preference;
+import org.apereo.portal.portlets.StringListAttribute;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class UserAccountHelperTest {
+
+    private static Set<String> CURRENT_ATTRIBUTE_NAMES;
+
+    private static List<Object> CURRENT_ATTRIBUTE_VALUES;
+
+    private static final String TEST_ATTRIBUTE_NAME = "attr_name_0";
+
+    private static final String TEST_USER_NAME = "Jane Doe";
+
+    private static final String TEST_USER_LOGIN_TOKEN = "16526520982";
+
+    private UserAccountHelper userAccountHelper;
+
+    @Mock private ILocalAccountDao accountDao;
+
+    @Mock private ILocalAccountPerson accountPerson;
+
+    static {
+        CURRENT_ATTRIBUTE_NAMES =
+                Collections.unmodifiableSet(
+                        new HashSet<>(
+                                Arrays.asList(
+                                        new String[] {
+                                            "attr_name_0",
+                                            "attr_name_1",
+                                            "attr_name_2",
+                                            "attr_name_3"
+                                        })));
+
+        CURRENT_ATTRIBUTE_VALUES =
+                Collections.unmodifiableList(
+                        new ArrayList<>(Arrays.asList(new String[] {"attr_value_0"})));
+    }
+
```

Comment:
Interesting approach. I like your use of unmodifiedSet/List wrappers on your static collection members.